src/charm.py

```diff
@@ -228,6 +248,15 @@ def _validate(self):
         if not self._state.schema_ready:
             raise ValueError("admin:temporal relation: schema is not ready")
 
+        if self.config["auth-enabled"]:
+            if not self._state.openfga:
+                raise ValueError("openfga:temporal relation not ready")
+            missing_params = self._check_missing_openfga_params()
+            if len(missing_params) > 0:
+                raise ValueError(f"openfga:missing parameters {missing_params!r}")
```

Comment:
Is the "r" in the end of the {missing_params!r} intended?

Reply:
Yes, it's used in f-strings to convert the value to its representation string (i.e. `value` would be output as `'value'`)